### DISCUSSION TITLE
Errors with some HTML snippets with tags (mdx 1.6.22)

### DISCUSSION THREAD

**Discussion by tnir** [mdx-js/mdx#1568] [Q&A] [answered]
Is this reasonable?

### OK
```markdown
<div>

{a: 0}
</div>
```
=> OK

### NG
```markdown
<div>
{a: 0}
</div>
```
=> **SyntaxError: No-Inline evaluations must call `render`.**

## Env

There is no differences in the following test environments:

- `@mdx-js/mdx` 1.6.22 (local)
- https://mdxjs.com/playground (as of 2021-06-01)

**Comment by ChristianMurphy** [comment, marked as answer]
It is, in MDX 1 an empty newline inside a JSX block allows non-JSX content to be mixed in.
In the first example, the JSX block has switched back to markdown, where `{a: 0}` is interpreted as plain text.
In the second example, the content is still inside JSX, and is interpreted as a JSX expression, and `a: 0` is invalid JS/JSX.

**Reply by tnir** [reply]
Oh, I see. Looks perfect! ❤️
